### PR TITLE
chore(ui): migrate Grid, GridColumn and GridRow components to Typescript

### PR DESCRIPTION
**Pull request by guoda-puidokaite**
# Summary

- Migrated the `Grid`, `GridColumn` and `GridRow` components to `TypeScript`, enhancing code quality, maintainability, and scalability.
- Utilised static typing to minimize runtime errors and improve the development experience.
- Removed deprecated `Grid`, `GridColumn` and `GridRow` `.js` components
- Refactored the component for cleaner, more maintainable code.

# Changes Made

**TypeScript Migration**:

- Converted `Grid`, `GridColumn` and `GridRow` component files to `TypeScript`.
- Added comprehensive type annotations.

**Cleanup:**

- Removed deprecated code. It wasn't needed for any `.js` components.

**Refactoring**:

- Made necessary adjustments to address TypeScript-detected issues.
- Further refactored for cleaner and more understandable code.

**Testing**:

- Added edge case tests: https://github.com/cloudoperators/juno/pull/595/commits/28dc6becae5fdd702d22b7ded014648d9bcfda7d
- Organised tests by what they check for better readability and maintenance.
- Verified component behaviour using `Storybook`.
- Updated tests to use `vitest`.

**Documentation:**

- Updated `Storybook` documentation.

**Release:**

- Generated a `changeset` file.

# Related Issues

- https://github.com/cloudoperators/juno/issues/229

# Testing Instructions

- Check deployed`Storybook`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/components/Grid/Grid.stories.tsx [hunk 1]
```diff
@@ -54,7 +55,7 @@ export const Default = {
       <GridRow key="2">
         <GridColumn>Column</GridColumn>
         <GridColumn cols={3}>Column cols-3</GridColumn>
-        <GridColumn cols={5}>Column cols-6</GridColumn>
```
Comment (TilmanHaupt) [inline packages/ui-components/src/components/Grid/Grid.stories.tsx:57]: cool that you found that. ^^

**Review by TilmanHaupt — APPROVED** [review]
lgtm :)

File: packages/ui-components/src/components/Grid/Grid.component.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+// Styles to apply when the 'auto' prop is true, overriding default grid column properties
+const autoStyles = {
+  "--grid-column-flex-grow": "1",
+  "--grid-column-flex-shrink": "0",
+  "--grid-column-flex-basis": "0",
+  "--grid-column-default-width": "auto",
+}
+
+export interface GridProps extends React.HTMLAttributes<HTMLDivElement> {
```
Comment (barsukov) [inline packages/ui-components/src/components/Grid/Grid.component.tsx:16]: Is this reflecting our challenge? 
https://github.com/cloudoperators/juno/issues/583

**Review by barsukov — APPROVED** [review]
HERE WE GO

**Review by andypf — APPROVED** [review]
LGTM
